listen.go

```diff
@@ -149,78 +138,98 @@ func (l *listener) handleIncoming() {
 	defer wg.Done()
 
 	for {
-		maconn, err := l.Listener.Accept()
+		conn, err := l.Listener.Accept()
 		if err != nil {
 			if l.catcher.IsTemporary(err) {
 				continue
 			}
-
-			l.incoming <- connErr{err: err}
+			l.incoming <- connOrErr{err: err}
 			return
 		}
 
-		log.Debugf("listener %s got connection: %s <---> %s", l, maconn.LocalMultiaddr(), maconn.RemoteMultiaddr())
-
-		if l.filters != nil && l.filters.AddrBlocked(maconn.RemoteMultiaddr()) {
-			log.Debugf("blocked connection from %s", maconn.RemoteMultiaddr())
-			maconn.Close()
+		if l.filters != nil && l.filters.AddrBlocked(conn.RemoteMultiaddr()) {
+			log.Debugf("blocked connection from %s", conn.RemoteMultiaddr())
+			conn.Close()
 			continue
 		}
-		// If we have a wrapper func, wrap this conn
-		if l.wrapper != nil {
-			maconn = l.wrapper(maconn)
-		}
+
+		log.Debugf("listener %s got connection: %s <---> %s", l, conn.LocalMultiaddr(), conn.RemoteMultiaddr())
 
 		wg.Add(1)
 		go func() {
 			defer wg.Done()
-			if l.protec != nil {
-				pc, err := l.protec.Protect(maconn)
+
+			ctx, cancel := context.WithTimeout(l.ctx, ConnAcceptTimeout)
```

Comment:
We should consider setting a `ConnAcceptTimeout` deadline on the connection itself to avoid leaking go routines.

Reply:
As with dialing, the connection will be closed when this expires so ignore this.